### DISCUSSION TITLE
Can't start grapher using local setup guide - ER_NO_SUCH_TABLE error

### DISCUSSION THREAD

**Discussion by MahmoudHamdy02** [owid/owid-grapher#1963] [Q&A]
When following the [local setup guide,](https://github.com/owid/owid-grapher/blob/master/docs/docker-compose-mysql.md) I can't access localhost:3030 or the admin page. I get ER_NO_SUCH_TABLE error on both pages. Docker tab in tmux shows that the DB is loaded.

Here are a few screenshots:
![image](https://user-images.githubusercontent.com/90795679/221730473-d78c122c-151e-4753-8f19-67a24d52331d.png)
![image](https://user-images.githubusercontent.com/90795679/221730618-fa239d01-d334-43a5-8c0d-9fab4370d445.png)
![image](https://user-images.githubusercontent.com/90795679/221730674-9ccefcb3-e7b3-444c-a500-d45858285dd4.png)

I have tried removing all volumes and temp data and downloading the DB again, but that didn't work.
I have tried these commands: `yarn runDbMigrations` and `yarn refresh` that I found while searching about this issue, but they didn't help either.

I'm using Ubuntu 22.04, Firefox 110.0, latest commit of grapher repository.

Thanks!

**Comment by marcelgerber** [comment]
Hi @MahmoudHamdy02, could you run the following in terminal?
```
cd [path-to-your-grapher-repository]
ls -la ./tmp-downloads/owid_metadata.sql.gz
```

What is the date it shows there? Is it a recent one, or is it pretty old?


If my assumption is true, and this is indeed a pretty old version, then you can do one of the following:
1. `yarn lerna run build && yarn buildTsc && yarn runDbMigrations`, **while** `make up` is running in the background (this will migrate your existing data)
2. `make refresh`, **while** `make up` is running in the background (this will override your existing data)

**Reply by MahmoudHamdy02** [reply]
Hello @marcelgerber!

This is the result from the `ls` command:
`-rw-rw-r-- 1 mahmoud mahmoud 61838743 Feb 28 03:11 ./tmp-downloads/owid_metadata.sql.gz`

It's today's date so it looks like a recent one to me. 

I tried running `yarn lerna run build && yarn buildTsc && yarn runDbMigrations`, but I got a similar error:

![image](https://user-images.githubusercontent.com/90795679/221940218-533e962e-24ad-4b4d-8d79-c0aa4c34ca4a.png)

![image](https://user-images.githubusercontent.com/90795679/221940479-3d4b5cb6-afc7-4c7c-a816-180aae0b4fe9.png)

![image](https://user-images.githubusercontent.com/90795679/221940741-79bcc46c-7d65-4b75-b200-1f61d1c02131.png)

**Reply by marcelgerber** [reply]
Oh! In that case, it looks like the initial import didn't succeed. Maybe you stopped the Docker runner before the import was complete, or something was going awry with the credentials.

To un-screw the database, try running as I described above:
> `make refresh`, while `make up` is running in the background (this will override your existing data)

For this to work, you will have to have `mysql-client` installed on your machine (ideally mysql 8.0, but 5.7 may also work). How you do that depends on the distribution you're using.

**Comment by DamilolaDecarls** [comment]
Update: 08.03.2024. I am currently getting this error. The error I get is similar to the error @MahmoudHamdy02 got <See attached screenshot>
![Screenshot 2024-03-08 at 19 23 13](https://github.com/owid/owid-grapher/assets/44199406/d85ee236-3935-461a-8559-9e92695553e7)


I have attempted the `make refresh.` It does not work. 

ADDITIONAL CONTEXT

<img width="1175" alt="Screenshot 2024-03-08 at 19 26 37" src="https://github.com/owid/owid-grapher/assets/44199406/cb3fd814-bb93-4703-a3cd-d3ce03bf5381">

I'd appreciate it if anyone could help me find a solution to this.

**Reply by marcelgerber** [reply]
Hi!
Can you try running `make refresh` while, in another tab or terminal window, `make up` is running?

**Comment by toni-sharpe** [comment]
See: https://github.com/owid/owid-grapher/pull/3636

=====================================

~Possibly unrelated, and I closed it after a couple of weeks, but I did [try to improve the docs further here](https://github.com/owid/owid-grapher/pull/3226). But I didn't include this yet, which is how I got it running.~

~I see in the original error: "A database with the name 'grapher' already exists. exiting" - my guess: because it exits it's never making the tables - ~IIRC~ [edit] I have now checked the logs I grabbed at the time - my eventual resolution to this was the complete removal of all Docker - my feeling is as this was a machine on which I have had several projects there are dockers lying around blocking stuff, and, that company didn't get us Docker for Desktop licenses. Even stopping everything in that application didn't find everything that the terminal commands did.~

~* Stop everything in Docker for Desktop - all containers, images, volumes and builds~
~* Close/quit that software~
~* $ `docker stop $(docker ps -a -q)` <-- stop everything in the command line~
~* $ `docker rm $(docker ps -a -q)` <-- remove everything in the command line~
~* Remove owid directories~
~* Restart machine~
~* `vim ~/.docker/config.json` and change `credsStore` into `credStore` if you need to~
~* Restart Docker for Desktop~
~* Reclone owid~
~* Follow the setup instructions~

~Note that `yarn startAdminServer` [has also helped to get the localhosts running - credit where it's due](https://github.com/owid/owid-grapher/discussions/3122#discussioncomment-8200683)~
